### PR TITLE
fix(ui): fetch clubs from api

### PR DESCRIPTION
**Pull request by prajjwalkapoor**

### CODE REVIEW

**Comment by prajjwalkapoor** [comment]
@Samy-33 you can review this

File: ui/src/components/ClubsCarousel/index.tsx [hunk 1]
```diff
@@ -1,37 +1,46 @@
...
-    name: 'Badminton',
-    category: 'Sports',
-    logoUrl: dummyClubLogo,
-  },
-];
+import {useEffect, useState} from 'react';
+import {useNavigate} from 'react-router-dom';
 
 export const ClubsCarousel: React.FC = () => {
+  interface ClubsCarouselProps {
```
Comment (Samy-33) [inline ui/src/components/ClubsCarousel/index.tsx:8]: These are not Carousel props, so naming them so doesn't make sense. Maybe `ClubsCarouselData` sounds better?

File: ui/src/components/ClubsCarousel/index.tsx [hunk 2]
```diff
@@ -1,37 +1,46 @@
...
+    name: string;
+    logo: string;
+    category: string;
+  }
+  const [data, setData] = useState<ClubsCarouselProps[]>([]);
+  const token = window.sessionStorage.getItem('token');
+  const CLUBS_API_URL = 'http://127.0.0.1:8000/clubs/';
```
Comment (Samy-33) [inline ui/src/components/ClubsCarousel/index.tsx:15]: Should not be hardcoded. Look at https://github.com/BitByte-TPC/gymkhana/blob/294716bc464488827e6d9e9d9aa55cb32ee02152/ui/src/api/googleOauth.ts#L34 for reference.

File: ui/src/components/ClubsCarousel/index.tsx [hunk 3]
```diff
@@ -52,7 +61,7 @@ export const ClubsCarousel: React.FC = () => {
       <Carousel responsive={responsive} className={styles.carousel}>
         {data.map((club, i) => (
```
Comment (Samy-33) [inline ui/src/components/ClubsCarousel/index.tsx:62]: rename `i` to `index` for better readability.

**Review by Samy-33 — CHANGES_REQUESTED** [review]

**Review by Samy-33 — APPROVED** [review]
LGTM

**Pull request merged**